components/hab/src/command/studio/enter.rs

```diff
@@ -272,3 +295,89 @@ mod inner {
         true
     }
 }
+
+#[cfg(test)]
+mod tests {
+    use super::cache_ssl_cert_file;
+    use std::fs::File;
+    use tempfile::TempDir;
+
+    #[test]
+    fn cache_ssl_cert_file_caches_file() -> std::io::Result<()> {
+        let cert_name = "ssl-test-cert.pem";
+        let cert_cache_dir = TempDir::new()?.into_path();
+        let ssl_cert_dir = TempDir::new()?;
+        let ssl_cert_filepath = ssl_cert_dir.path().join(cert_name);
+        File::create(&ssl_cert_filepath)?;
+
+        cache_ssl_cert_file(&ssl_cert_filepath.to_str().unwrap(), &cert_cache_dir);
+        assert!(cert_cache_dir.join(cert_name).exists());
+
+        Ok(())
+    }
+
+    #[test]
+    fn cache_ssl_cert_file_replaces_already_cached() -> std::io::Result<()> {
+        let cert_name = "ssl-test-cert.pem";
+
+        let cert_cache_dir = TempDir::new()?.into_path();
+        let cached_cert = cert_cache_dir.join(&cert_name);
+        File::create(&cached_cert)?;
+
+        let ssl_cert_dir = TempDir::new()?;
+        let ssl_cert_filepath = ssl_cert_dir.path().join(&cert_name);
+        std::fs::write(&ssl_cert_filepath, "new cert from environment")?;
+
+        cache_ssl_cert_file(&ssl_cert_filepath.to_str().unwrap(), &cert_cache_dir);
+
+        let contents = std::fs::read_to_string(&cached_cert)?;
+
+        assert_eq!(contents, "new cert from environment");
+
+        Ok(())
+    }
+    #[test]
+    fn cache_ssl_cert_file_invalid_file() -> std::io::Result<()> {
+        let cert_cache_dir = TempDir::new()?.into_path();
+
+        let non_existant_file_name = "i_shouldnt_exist";
+        let non_existant_file = TempDir::new()?.path().join(non_existant_file_name);
+
+        cache_ssl_cert_file(non_existant_file.to_str().unwrap(), &cert_cache_dir);
+        assert!(!cert_cache_dir.join(non_existant_file_name).exists());
+
+        Ok(())
+    }
+
+    #[test]
+    fn cache_ssl_cert_file_cert_file_is_dir() -> std::io::Result<()> {
+        let cert_cache_dir = TempDir::new()?.into_path();
+        let ssl_cert_dir = TempDir::new()?.into_path();
+        let needs_a_name = "i_am_a_directory";
+        std::fs::create_dir(ssl_cert_dir.join(needs_a_name))?;
+
+        cache_ssl_cert_file(&ssl_cert_dir.to_str().unwrap(), &cert_cache_dir);
+
+        assert!(!cert_cache_dir.join(needs_a_name).exists());
+
+        Ok(())
+    }
+
+    #[test]
+    fn cache_ssl_cert_file_invalid_cert_cache_dir() -> std::io::Result<()> {
```

Comment:
Sorry, I'm not sure what I was thinking with suggesting this particular test. Since `cache_ssl_cert_file` doesn't report any errors, there's nothing observable that could happen with passing an invalid `cert_cache_dir`. We can probably remove this test altogether.